hardware_device_context.go

```diff
@@ -0,0 +1,37 @@
+package astiav
+
+//#cgo pkg-config: libavutil libavcodec
+//#include <libavcodec/avcodec.h>
+//#include <libavutil/hwcontext.h>
+import "C"
+import (
+	"unsafe"
+)
+
+// https://github.com/FFmpeg/FFmpeg/blob/n5.0/libavutil/hwcontext.h#L141
+type HardwareDeviceContext struct {
+	c *C.AVBufferRef
```

Comment:
Could you try replacing `C.AVBufferRef` with `C.struct_AVHWDeviceContext` and let me know whether that still works?

Reply:
this does not work
`
.\hardware_device_context.go:35:51: cannot use _cgo0 (variable of type **_Ctype_struct_AVHWDeviceContext) as **_Ctype_struct_AVBufferRef value in argument to _Cfunc_av_buffer_unref`

Reply:
that was worth a try 👍